Subject: appliance: enable bash's Process Substitution feature
From: Olaf Hering olaf@aepfle.de Wed Sep 15 12:20:42 2021 +0200
Date: Wed Sep 15 12:37:08 2021 +0100:
Git: 9db0c98c99090e601d856c6795544f6967e6155f

bash can read input from a spawned process, and even provide input to
such process. This feature relies on /dev/fd/ being present. In the
past udev silently created this symlink, so this bash feature worked
more or less by accident. With recent systemd versions, such as 246
which is included in Leap 15.3, the symlink is not created anymore. As
a result scripts, such as /sbin/dhclient-script, fail to work
properly.

This symlink should have been created in version 1 of this variant of /init.

https://bugzilla.opensuse.org/show_bug.cgi?id=1190501

Signed-off-by: Olaf Hering <olaf@aepfle.de>

diff --git a/appliance/init b/appliance/init
index b1c4d09ea..cdc39c3b9 100755
--- a/appliance/init
+++ b/appliance/init
@@ -72,6 +72,7 @@ fi
 
 # devtmpfs is required since udev 176
 mount -t devtmpfs /dev /dev
+ln -s /proc/self/fd /dev/fd
 mkdir -p /dev/pts
 mount -t devpts /dev/pts /dev/pts
 mkdir -p /dev/shm
